src/common/_io.py

```diff
@@ -0,0 +1,123 @@
+"""
+module: _io.py
+
+Module that contains functions responsible for moving, creating, writing and
+deleting files in the current directory.
+"""
+import json
+from pathlib import Path
+from typing import List, Optional
+
+from PIL import Image
+
+from src.common.errors import NotCoordArrayError
+from src.guards.arrays_guards import is_list_of_crop_selections
+from src.structs.selections import ImageCropSelection
+
+
+def load_tiff_images(
+    img_dir: str | Path,
+) -> List[Image.Image]:
+    """Loads images from given directory. Returns a list object
+
+    Parameters
+    ----------
+    img_dir : str | Path
+       directory path containing image paths
+
+    Returns
+    -------
+    List[PIL.TiffImagePlugin.TiffImageFile]
+        List of tuples that contains the image path and the PIL image object
+
+    Raises
+    ------
+    FileNotFoundError
+        Raised if the 'img_dir' does not exist
+    TypeError
+        Raised if images are not in `tiff` format
+        Raised if `img_dir` is not str or Path type
+    """
+
+    # type checking
+    # -- if str type, convert to Path type
+    accepted_types = (str, Path)
+    if not isinstance(img_dir, accepted_types):
+        _type = type(img_dir)
+        raise TypeError(f"'fpath' must str or path type not {_type}")
+    if isinstance(img_dir, str):
+        img_dir = Path(img_dir)
+    if not img_dir.exists():
+        raise FileNotFoundError(f"Unable to find: {str(img_dir)}")
+    if img_dir.suffix == "tiff":
+        raise TypeError(f"Image must be .tiff format not {img_dir.suffix}")
+
+    # grab all .tiff images within given image directory
+    tiff_paths = img_dir.glob("*.tiff")
+
+    # loading images into memory
+    # -- List[Tuple(path/to/image, PIL Image Object)]
+    loaded_images = []
+    for tiff_path in tiff_paths:
+        tiff_img = Image.open(tiff_path)
+        loaded_images.append(tiff_img)
```

Comment:
This has the potential to get _huge_.

I think it is ok to merge this in for now as is, but it is worth thinking about in the future if loading all images is necessary. Can you think of a way to not need all images in memory at any point?

Reply:
Yes, I was thinking about this when I was reviewing it myself. I plan on changing this into a python generator object instead of returning a list. ( it is on the works!)